pypicloud/cache/base.py

```diff
@@ -170,6 +172,71 @@ def distinct(self):
         """
         raise NotImplementedError
 
+    def search(self, criteria, query_type):
+        """
+        Perform a search from pip
+
+        Parameters
+        ----------
+        criteria : dict
+            Dictionary containing the search criteria. Pip sends search criteria
+            for "name" and "summary" (typically, both of these lists have the
+            same search values).
+
+            Example:
+            {
+                "name": ["value1", "value2", ..., "valueN"],
+                "summary": ["value1", "value2", ..., "valueN"]
+            }
+
+        query_type : str
+            Type of query to perform. By default, pip sends "or".
+
+        """
+        name_queries = criteria.get('name', [])
+        summary_queries = criteria.get('summary', [])
+        packages = []
+        packages_found = set()
+
```

Comment:
This method doesn't respect the `query_type`. To support that, I'd suggest a slightly different mechanism for doing the matching. Something like:
```python
def create_matcher(queries, query_type):
    queries = [q.lower() for q in queries]
    if query_type == 'or':
        return lambda x: any((q in x.lower() for q in queries))
    else:
        return lambda x: all((q in x.lower() for q in queries))

match_name = create_matcher(name_queries, query_type)
match_summary = create_matcher(summary_queries, query_type)
```

And then below you can just use `if match_name(package.name): ...`